packages/pipeline-editor/src/utils.ts

```diff
@@ -52,4 +55,69 @@ export default class Utils {
 
     return template;
   }
+
+  /**
+   * Read the version of a Pipeline. If no version is found return 0
+   */
+  static getPipelineVersion(pipelineDefinition: any): number {
+    const version: number =
+      +this.getPipelineAppdataField(
+        pipelineDefinition.pipelines[0],
+        'version'
+      ) || 0;
+
+    return version;
+  }
+
+  /**
+   * Read an application specific field from the pipeline definition
+   * (e.g. pipelines[0][app_data][fieldName])
+   */
+  static getPipelineAppdataField(node: any, fieldName: string): string {
+    if (this.hasPipelineAppdataField(node, fieldName)) {
+      return node['app_data'][fieldName] as string;
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Check if an application specific field from the pipeline defintion exists
+   * (e.g. pipelines[0][app_data][fieldName])
+   */
+  static hasPipelineAppdataField(node: any, fieldName: string): boolean {
+    let isPresent = false;
+    if (node['app_data']) {
+      if (node['app_data'][fieldName]) {
```

Comment:
if the `fieldName` is present but has a value of `0`, `false`, or an empty string, this function may not behave as intended. although, the field is present this would return false.

depending on what you are trying to achieve a better approach is to check the just the existence of a field regardless of it's value would be:

```javascript
  return Object.prototype.hasOwnProperty.call(node, 'app_data') &&
    Object.prototype.hasOwnProperty.call(node['app_data'], fieldName);
```